pkg/netpol/eval/internal/k8s/representative_selectors.go

```diff
@@ -0,0 +1,120 @@
+		return true, nil
+	}
+	// otherwise, check requirements equality
+	selector1, err := v1.LabelSelectorAsSelector(ruleSelector)
+	if err != nil {
+		return false, err
+	}
+	if selector1.Empty() { // empty rule matches everything
+		return true, nil
+	}
+	selector2, err := v1.LabelSelectorAsSelector(repSelector)
+	if err != nil {
+		return false, err
+	}
+	// Requirements() returns sorted by key list;
+	// since `v1.LabelSelectorAsSelector` builds a new `labels.selector` by using `Requirements.Add` which sorts the requirements slice by key.
+	// and `Requirements()` returns this sorted list
+	// links:
+	// v1.LabelSelectorAsSelector:
+	// https://github.com/kubernetes/apimachinery/blob/dc7e034c86479d49be4b0eefad307621e10caa0e/pkg/apis/meta/v1/helpers.go#L34
+	// Requirements.Add : https://github.com/kubernetes/apimachinery/blob/d7e1c5311169d5ece2db0ae0118066859aa6f7d8/pkg/labels/selector.go#L373
+	requirements1, _ := selector1.Requirements() // sorted
+	requirements2, _ := selector2.Requirements() // sorted
+	if len(requirements1) != len(requirements2) {
+		return false, nil
+	}
+	for i := range requirements1 {
+		// requirements1[i].Equal(requirements2[i]) returns false if the values are not in same order
+		// (stringslices.Equal returns true only if two slices have same length and same order of items)
+		// however, `Requirement.String` sorts the values using `safeSort`, so will compare by string
+		// link to Requirement.String() :
+		// https://github.com/kubernetes/apimachinery/blob/d7e1c5311169d5ece2db0ae0118066859aa6f7d8/pkg/labels/selector.go#L310
+		str1 := requirements1[i].String()
+		str2 := requirements2[i].String()
+
+		// handling special case of one requirement has `In` operator with 1 value, and other requirement has "=" operator (from matchLabel)
+		// for example : req1 := (app in x), req2 := (app=x) >> we expect a full match, however,
+		// labels pkg treats them as different requirements
+		// i.e. requirements1[i].Equal(requirements2[i]) and requirements1[i].String() != requirements2[i].String() return false in this case
+		// requirement.String() : returns <key>=<values> string for "Equals" operator and returns (<key> in (<values));
+		// so, in case on requirement is "in" with one value only , will convert its string to the <key>=<values> format to get correct result
+		if newStr1 := handleRequirementWithInOpAndSingleValue(requirements1[i]); newStr1 != "" {
+			str1 = newStr1
+		}
+		if newStr2 := handleRequirementWithInOpAndSingleValue(requirements2[i]); newStr2 != "" {
+			str2 = newStr2
+		}
+		if str1 != str2 {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+// handleRequirementWithInOpAndSingleValue returns a <key>=<val> string format if the input Requirement is with In operator and single value
+func handleRequirementWithInOpAndSingleValue(req labels.Requirement) string {
+	if req.Operator() == selection.In && len(req.Values()) == 1 {
+		return req.Key() + "=" + req.Values().List()[0]
+	}
+	return ""
+}
```

Comment:
the name `VariantFromLabelsSelector` is misleading.. consider renaming to UniqueKeyFromLabelsSelector

Reply:
done